scripts/m3dbnode_bootstrapped.sh

```diff
@@ -0,0 +1,51 @@
+#!/bin/bash
+
+# Exits with code 0 if either:
+# - There is no placement
+# - The current host is not in the placement
+# - The current is in the placement AND all it's shards are AVAILABLE
+#
+# Exits 1 otherwise.
+
+set -exo pipefail
+
+COORDINATOR_PORT=${COORDINATOR_PORT:-7201}
+ENDPOINT="http://localhost:${COORDINATOR_PORT}/api/v1/placement"
+HOSTNAME=${HOSTNAME:-$(hostname)}
+
+TMPFILE=$(mktemp)
+
+function cleanup() {
+  rm -f "$TMPFILE"
+}
+
+trap cleanup EXIT
+
+curl -sSf -o "$TMPFILE" "$ENDPOINT"
+RES=$?
+
+# Curl exits 22 for 400+ status code. Note this leaves us vulnerable to saying
+# bootstrapped if our script makes a bad request and must use caution when
+# modifying the script or the coordinator placement endpoint.
+if [[ "$RES" -eq 22 ]]; then
+  echo "Received 400"
+  exit 0
+fi
+
+# TODO(schallert): should a leaving node be considered bootstrapped?
```

Comment:
"Leaving" status is on a shard-by-shard basis and doesn't have much relation to whether the node is bootstrapped or not